File: assistant-backend-demo/src/main/scala/config/IntegrationConfig.scala
package config

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.stereotype.Component

import scala.beans.BeanProperty

@Component
@ConfigurationProperties(prefix = "integration")
class IntegrationConfig {
  @BeanProperty
  var smartdocsAiAssistantUrl: String = _

  @BeanProperty
  var smartdocsAiAssistantApiKey: String = _
}

File: assistant-backend-demo/java/src/main/java/com/bgl/ai/chatbot/config/IntegrationConfig.java
package com.bgl.ai.chatbot.config;

import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.stereotype.Component;

@Component
@ConfigurationProperties(prefix = "integration")
public class IntegrationConfig {

    private String smartDocsAiAssistantUrl;
    private String smartDocsAiAssistantApiKey;

    public String getSmartDocsAiAssistantUrl() {
        return smartDocsAiAssistantUrl;
    }

    public void setSmartDocsAiAssistantUrl(String smartDocsAiAssistantUrl) {
        this.smartDocsAiAssistantUrl = smartDocsAiAssistantUrl;
    }

    public String getSmartDocsAiAssistantApiKey() {
        return smartDocsAiAssistantApiKey;
    }

    public void setSmartDocsAiAssistantApiKey(String smartDocsAiAssistantApiKey) {
        this.smartDocsAiAssistantApiKey = smartDocsAiAssistantApiKey;
    }
}
